apps/api/scripts/backfill-agent-score.ts

```diff
@@ -0,0 +1,564 @@
+import {
+  competitionAgents,
+  competitions,
+} from "@/database/schema/core/defs.js";
+import {
+  agentScore,
+  agentScoreHistory,
+} from "@/database/schema/ranking/defs.js";
+
+interface Args {
+  competitions: string;
+  execute?: boolean;
+  exclude?: string;
+}
+
+/**
+ * Prompt user for confirmation
+ */
+async function promptConfirmation(message: string): Promise<boolean> {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  return new Promise((resolve) => {
+    rl.question(`${message} (y/n): `, (answer) => {
+      rl.close();
+      resolve(answer.toLowerCase() === "yes" || answer.toLowerCase() === "y");
+    });
+  });
+}
+
+/**
+ * Check for missing agent score history entries
+ */
+async function checkMissingAgentScores(competitionId: string) {
+  // Get active agents in the competition
+  const activeAgents = await db
+    .select({
+      agentId: competitionAgents.agentId,
+    })
+    .from(competitionAgents)
+    .where(
+      and(
+        eq(competitionAgents.competitionId, competitionId),
+        eq(competitionAgents.status, "active"),
+      ),
+    );
+
+  // Get existing agent score history for this competition
+  const existingHistory = await db
+    .select({
+      agentId: agentScoreHistory.agentId,
+    })
+    .from(agentScoreHistory)
+    .where(eq(agentScoreHistory.competitionId, competitionId));
+
+  const existingAgentIds = new Set(existingHistory.map((h) => h.agentId));
+  const missingAgents = activeAgents.filter(
+    (a) => !existingAgentIds.has(a.agentId),
+  );
+
+  return {
+    totalActiveAgents: activeAgents.length,
+    missingAgents: missingAgents.map((a) => a.agentId),
+  };
+}
+
+/**
+ * Backfill agent scores for a specific competition
+ */
+async function backfillCompetitionScores(competitionId: string) {
+  console.log(
+    chalk.blue(`\nBackfilling scores for competition ${competitionId}...`),
+  );
+
+  const competition = await db
+    .select()
+    .from(competitions)
+    .where(eq(competitions.id, competitionId))
+    .limit(1);
+
+  if (!competition[0]) {
+    console.log(chalk.red(`Competition ${competitionId} not found`));
+    return;
+  }
+
+  const comp = competition[0];
+  console.log(chalk.gray(`Competition: ${comp.name}`));
+
+  // Get ALL active agents in this competition (not just those with leaderboard entries)
+  const activeAgents = await db
+    .select({
+      agentId: competitionAgents.agentId,
+    })
+    .from(competitionAgents)
+    .where(
+      and(
+        eq(competitionAgents.competitionId, competitionId),
+        eq(competitionAgents.status, "active"),
+      ),
+    );
+
+  if (activeAgents.length === 0) {
+    console.log(chalk.yellow(`No active agents found, skipping`));
+    return;
+  }
+
+  console.log(chalk.gray(`Processing ${activeAgents.length} active agents`));
+
+  // Get the leaderboard for this competition (might be incomplete)
+  const leaderboard =
```

Comment:
So we need to backfill the leaderboard data first, right?

Reply:
correct. first, via the `backfill-stats` script, we'll insert in to the leaderboard tables for all "ended" comps: https://github.com/recallnet/js-recall/blob/7b92c23c26a1a4325574a62e002a8b81e2fac0c1/apps/api/scripts/backfill-stats.ts#L204

then, delete the bad comps via the `delete-competitions` script. lastly, backfill/recalculate the agent scores and history via this PR's script.